docs/language.md

```diff
@@ -302,6 +305,61 @@ macro filter(data, func) = (
 );
 ```
 
+## Custom Macros
+
+The operations listed above are macros that are loaded by default. However, users can also define their own macros. Macro definitions must come after [type aliases](#typename-aliasing)  and precede the Weld expression that is to be compiled:
+
+```
+# A macro to add values.
+macro doubleAdd(a, b) = (
+  a + a + b + b
+);
+
+|v1: i32, v2: i32|
+  # Expands to v1 + v1 + v2 + v2
+  doubleAdd(v1 ,v2)
+```
+
+Macros are _hygenic_, so variable names defined within a macro will never clash (and by extension cannot be accessed from outside the macro expansion).
+Since macros do not require types in their parameters, they are a useful way to leverage type inference and write templatized Weld code.
```

Comment:
Do top-level Weld lambdas have to have typed parameters? If so this seems to need a clarification (i.e. the entire program can't just be a macro call, even though a macro is technically a complete Weld program).

Reply:
Good point, I'll just delete this comment.